enhancements/local-storage/automatic-detection.md

```diff
@@ -0,0 +1,157 @@
+---
+title: automatic-detection-of-disks-in-Local-Storage-Operator
+authors:
+  - "@aranjan"
+reviewers:
+  - "@jrivera"
+  - "@jsafrane"
+  - "@hekumar"
+  - "@chuffman"
+  - "@rojoseph"
+  - "@sapillai"
+approvers:
+  - "@jrivera"
+  - "@jsafrane"
+  - "@hekumar"
+  - "@chuffman"
+creation-date: 2020-01-21
+last-updated: 2020-01-21
+status: implementable
+---
+
+# Automatic detection of Disks in Local-Storage-Operator 
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+
+## Summary
+
+When working with bare-metal OpenShift clusters, due to the absence of storage-provisioners like that in the cloud there needs a lot of manual work to be done for consuming local storage from nodes. The idea is to automate the manual that is required for consuming local-storage from the OpenShift cluster.
+
+## Motivation
+
+For consuming local storage from the nodes available in the OpenShift cluster a good amount of manual work has to be done for creating local PVs out of it, making hot-swapping of disks tough.
+
+### Goals
+
+- Automatically detection of available disks from nodes which can be used as PVs for OpenShift-cluster.
+- Should respond to the attach/detach events of the disks/devices.
+- Should have options for excluding a particular kind of disks.
+
+### Non-Goals
+
+## Proposal
+
+`local-storage-operator`is already capable of consuming local disks from the nodes and provisioning PVs out of it, but the disk/device paths needs to explicitly specified in the `LocalVolume` CR. The idea is to bring in one more layer of CR above it called `AutoDetectVolumes`.
+The controller for this CR will be responsible for automatically detecting disks/devices from the available nodes and creating/managing the life cycle of `LocalVolumes` which are governed by the `AutoDetectVolumes`.s
+
+### Risks and Mitigations
+
+N/A
```

Comment:
Risks:
* Detecting disks that contain data and/or are in-use
* Ensuring disks aren't re-detected as new or otherwise destroyed if their device path changes

Reply:
Thanks for the suggestion added it :)

Reply:
For the first risk, should checking whether device has partitions and/or device has a filesystem suffice, to ensure that device is already in use or contain data.